containers/release-candidate/README.md

```diff
@@ -21,9 +21,9 @@ To improve reproducibility and to minimize the effects and variability of a
 users environment, the Daffodil release container should be used to create
 release candidates.
 
-To build the Daffodil release candidate container image:
+To build or update the Daffodil release candidate container image:
 
-    podman build -t daffodil-release-candidate /path/to/daffodil.git/containers/release-candidate/
+    podman build -t daffodil-release-candidate https://github.com/apache/daffodil.git#main:containers/release-candidate
```

Comment:
This takes whatever the head of main is. Not a specific tag. Shouldn't this be taking a specific tag? 

Or the issue anyway is that the current state of the main branch is also subject to being "not quite the right thing" just as with a local change to the sandbox. 

So does this really improve things?

Reply:
Correct, it just uses whatever main is at the time of the build. There are two things I think this improves:

1) It makes it less likely to include local changes. If you have the latest daffodil repo but make local changes (e.g. like when I recently tested updating the container for Fedora 40), if I was doing a release it could be possible for me to forget to undo those changes and rebuild. By using a git URL, it at least ensures there are no local changes. 

2) It makes it easier for VS Code/SBT releases. The Daffodil VS Code and SBT extensions are in separate repos, so developers of those might not necessarily keep their Daffodil repo up to date, or they might not even have a local clone of the Daffodil repo at all. So they need to either make a fresh clone or remember to update their clone prior to building the container. Using a git URL avoids all this entirely. They just run `podman build` and it will fetch the latest source and build it. Note, I'm not suggesting anyone has done this, it just minimizes one potential source for errors.

You are right that this doesn't do anything to really know which version of the container was actually used for a specific build. But we already have that problem today. This didn't try to solve that, just tried to make things a little easier and avoid local changes.

I would eventually like this to all go away at some point. For example, some projects at ASF have infrastructure set up so that you just push a tag and that triggers everything on some build infrastructure. So ideally pushing a tag would cause a GitHub actions to build and run the container from that tagged commit. No need to have podman or build a container or anything. It would all happen remotely. Then we would know the tagged commit was used for building the container and building daffodil. That's a bit more effort though, especially since things like signing keys and SVN passwords are needed for the GitHub actions to do everything it needs to do. I think it's possible in theory, but a decent amount of work.

Reply:
What is example(s) of such an ASF project that has this "just push a tag" mechanism working?

Reply:
I don't know of a good way to find all of them, but searching through the INFRA Jira for "GPG_SIGNING_KEY" (I believe this is the secret that infra adds to GitHubActions), it looks like it's mostly projects that are part of [Apache Logging](https://logging.apache.org/).

Infra also has documentation about [Automated Release Signing](https://infra.apache.org/release-signing.html#automated-release-signing) and what they require before they will allow it.

The biggest issue is reproducibility--our builds are close to reproducible, and they are close enough that it's pretty easy to verify the differences are expected, but they aren't 100% bit-for-bit the same.

Note that if we only built/released source artifacts we could get it to work without a problem (those are already reproducible). But the binaries are much harder, and personally I think should be built as part of the release process, even though ASF doesn't require it. I've opened a PR with sbt-native-packager to fix our zip binaries: [sbt/sbt-native-packager#160](https://togithub.com/sbt/sbt-native-packager/pull/1602). The windows installer binaries I don't think we can ever make fully 100% reproducible. We'll have to figure out if ASF can make an exception if we ever want to go this route.

Also, ASF Infra is working on artifacts.apache.org which is something they are working on to ease the whole release process. I think it's still TBD how much that will affect our release process (e.g. will it include signing or not), but that's something to consider too. Though, I think that's a ways off.